Sources/PostgresNIO/New/PSQLError.swift

```diff
@@ -108,33 +108,33 @@ struct PostgresCastingError: Error, Equatable {
         }
 
         @usableFromInline
-        static let missingData = Self.init(.missingData)
+        public static let missingData = Self.init(.missingData)
         @usableFromInline
-        static let typeMismatch = Self.init(.typeMismatch)
+        public static let typeMismatch = Self.init(.typeMismatch)
         @usableFromInline
-        static let failure = Self.init(.failure)
+        public static let failure = Self.init(.failure)
     }
 
     /// The casting error code
-    let code: Code
+    public let code: Code
 
     /// The cell's column name for which the casting failed
-    let columnName: String
+    public let columnName: String
     /// The cell's column index for which the casting failed
-    let columnIndex: Int
+    public let columnIndex: Int
     /// The swift type the cell should have been casted into
-    let targetType: Any.Type
+    public let targetType: Any.Type
     /// The cell's postgres data type for which the casting failed
-    let postgresType: PostgresDataType
+    public let postgresType: PostgresDataType
     /// The cell's postgres format for which the casting failed
-    let postgresFormat: PostgresFormat
+    public let postgresFormat: PostgresFormat
     /// A copy of the cell data which was attempted to be casted
-    let postgresData: ByteBuffer?
+    public let postgresData: ByteBuffer?
 
     /// The file the casting/decoding was attempted in
-    let file: String
+    public let file: String
     /// The line the casting/decoding was attempted in
-    let line: Int
+    public let line: Int
     
     var description: String {
```

Comment:
You have moved this as intended into its own extension. I think we should remove it here.